### PR TITLE
Products store page

### PR DESCRIPTION
**Pull request by agrawalshanu708**
added filter bar
added sortbar
added products
added functionalities of all filters

### CODE REVIEW

**Review by k-rahul21 — APPROVED** [review]

File: src/components/filterbar/filterbar.css [hunk 1]
```diff
@@ -0,0 +1,66 @@
+.filter-container{
+  
+    display: flex;
+    background-color:#FFFFFF;
+    flex-direction: column;
+    /* align-items: center; */
```
Comment (muskaanshah) [inline src/components/filterbar/filterbar.css:6]: Remove commented code if not required

File: src/components/filterbar/filterbar.jsx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+             <input type="text" />
+             <input type="text" placeholder=""/>
+         </div>
+         <input type="range" min="1" max="100" />
+     </div>
+
+  <div className="accordian-item flex-vertical-outer ">
+  <div className="accordian-tap ">
+    <div className="filter-text">Categories</div>
+    {/* <MdKeyboardArrowRight className="arrow-right" size = "2rem"/>
```
Comment (muskaanshah) [inline src/components/filterbar/filterbar.jsx:38]: Remove commented code from all places if not required

File: src/reducer/filterreducer.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+         case "SORT_BY":
+            return {...state, sortby: action.payload}       
+        case "CLEAR":
+            return {...state,
+                sortby: null,
+                gender: null,
+                categoryName:{casual:false,formal:false,digital:false,sports:false,},
+                brandName :{apple:false,fossil:false,fastrack:false,titan:false},
+                discount: null
+            }
```
Comment (muskaanshah) [inline src/reducer/filterreducer.jsx:35]: Suggestion: Dont hard code it for every case. Make a common function.

File: src/utils/filterbygender.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+const filterByGender = (products,gender) => {
+  
+    switch (gender) {
+        case "SHOW_MEN":
+        return products.filter(item => item.category === "male")    
+        case "SHOW_WOMEN":
+        return products.filter(item => item.category === "female")
+        case "SHOW_KID":
+         return products.filter(item => item.category === "kid")   
+
```
Comment (muskaanshah) [inline src/utils/filterbygender.jsx:12]: Make a common function and pass value as parameter, hard coding is a wrong way.

File: src/utils/filterbydiscount.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+     case "ABOVE_10":
+     return products.filter(item => item.discount >= 10)    
+      case "ABOVE_20":
+      return products.filter(item => item.discount >= 20) 
+      case "ABOVE_30":
+     return products.filter(item => item.discount >= 30) 
+     case "ABOVE_40":
+         return products.filter(item => item.discount >= 40) 
+      case "ABOVE_50":
+          return products.filter(item => item.discount >= 50)
```
Comment (muskaanshah) [inline src/utils/filterbydiscount.jsx:14]: Make a common function and pass value as parameter, hard coding is a wrong way.

File: src/utils/getsort.jsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+   switch (sortby) {
+       case "PRICE_HIGH_TO_LOW":
+      return products.sort((a,b) => b.price-a.price)     
+       case "PRICE_LOW_TO_HIGH":
+           return products.sort((a,b) => a.price-b.price) 
+       case "RATING_HIGH_TO_LOW":
+         return products.sort((a,b) => b.rating["rate"]-a.rating["rate"])      
+       case "RATING_LOW_TO_HIGH":
+          return products.sort((a,b) => a.rating["rate"]-b.rating["rate"])  
+
```
Comment (muskaanshah) [inline src/utils/getsort.jsx:13]: Make a common function and pass value as parameter, hard coding is a wrong way.

**Review by muskaanshah — CHANGES_REQUESTED** [review]
Style changes- Put a style on hovering the buttons, user wont understand if it's a button or not. You can make the cursor as pointer when user hovers on the buttons.
Navbar's background is transparent because of which it is getting overlapped. Hard code the background colour for the navbar as the bg colour.
In a few cases, the filter function is not working properly.

**Review by therajatg — APPROVED** [review]
looks good

File: src/reducer/filterreducer.jsx (hunk 3 above)
Comment (agrawalshanu708) [reply src/reducer/filterreducer.jsx:35]: thanx muskan, I will make it

**Pull request merged**